src/bridge/IrcBridge.ts

```diff
@@ -828,6 +836,7 @@ export class IrcBridge {
             log.error(err.stack);
         });
 
+        await discoveringClientsPromise;
```

Comment:
out of band I realised that indeed things are fine, since we wait on `discoveringClientsPromise` after calling .sync(), but before trying any injects. TL;DR it's fine *now*.